java-squid/src/main/java/org/sonar/java/se/checks/SECheck.java

```diff
@@ -19,13 +19,19 @@
  */
 package org.sonar.java.se.checks;
 
+import com.google.common.collect.Multimap;
+import org.sonar.java.model.DefaultJavaFileScannerContext;
 import org.sonar.java.se.CheckerContext;
-import org.sonar.java.se.ConstraintManager;
+import org.sonar.java.se.constraint.ConstraintManager;
 import org.sonar.java.se.ProgramState;
 import org.sonar.plugins.java.api.JavaCheck;
+import org.sonar.plugins.java.api.JavaFileScanner;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
 import org.sonar.plugins.java.api.tree.Tree;
 
-public abstract class SECheck implements JavaCheck {
+import java.util.Map;
+
+public abstract class SECheck implements JavaCheck, JavaFileScanner {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) "org.sonar.plugins.java.api.JavaFileScanner" is a "org.sonar.plugins.java.api.JavaCheck" so "org.sonar.plugins.java.api.JavaCheck" can be removed from the extension list. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1939)